In [18]:
from tools_configs import *

# Read frequency word list

In [19]:

wordset_freq = {}
wordset = set()

with open("wordlists/chinese_charfreq_simpl_trad.txt", "r", encoding="utf-8") as fread:
    next(fread)
    contents = fread.read()

    wordset.update(contents)

    if "\n" in wordset:
        wordset.remove("\n")

    if " " in wordset:
        wordset.remove(" ")

    wordset_freq = {word: num + 1 for num, word in enumerate(contents.split("\n"))}

def find_freq(word):
    return wordset_freq[word] if word in wordset_freq else BIGNUM


def sort_by_freq(list_chars, reverse=False):
    if not list_chars:
        return []

    items = sorted(
        [(word, find_freq(word)) for word in list_chars], key=lambda x: (x[1], x[0]), reverse=reverse
    )

    return [word for word, order in items]

rads = Radicals()

rads.load_radical_data()

import json
try:
    with open('char_dict.json', "r", encoding="utf-8") as fread:
        char_dict = json.load(fread)
except:
    print(f'No file {'char_dict.json'}')

print(f'{len(char_dict)=}')
wordset = set()

with open("wordlists/chinese_charfreq_simpl_trad.txt", "r", encoding="utf-8") as fread:
    next(fread)
    contents = fread.read()

    wordset.update(contents)

    if "\n" in wordset:
        wordset.remove("\n")

    if " " in wordset:
        wordset.remove(" ")

    wordset_freq = {word: num + 1 for num, word in enumerate(contents.split("\n"))}

print(f'{len(wordset_freq)=}')
exiting_word = [word for word in wordset if word in char_dict]
print(f'{len(exiting_word)=}')

len(char_dict)=13052
len(wordset_freq)=12042
len(exiting_word)=11995


# Count
- No. Words by radical count
    
    - Most popular radicals

- Number and percents of words by radical count

- Check words agains top freqent words (top 12k)




In [20]:
chars_by_rad_count = {}
chars_by_radical = {}
most_freq_radicals = {}

for word in exiting_word:
    item = char_dict[word]

    components = item['components']

    count = 0
    for comp in components:
        if rads.is_radical_variant(comp):
            count += 1

            chars_by_radical.setdefault(rads.norminal(comp), [])
            chars_by_radical[rads.norminal(comp)].append(word)

    chars_by_rad_count.setdefault(count, [])
    chars_by_rad_count[count].append(word)

count_chars_by_radical = sorted([(key, len(value)) for key, value in chars_by_radical.items()], key = lambda x: x[1], reverse=True)
count_chars_by_rad_count = sorted([(key, len(value)) for key, value in chars_by_rad_count.items()], key = lambda x: x[1], reverse=True)

print('Most freq radical count')
for num, item in enumerate(count_chars_by_rad_count):
    print(f'{item[0]}\t{item[1]}')

print('Most freq radicals')
for num, item in enumerate(count_chars_by_radical):
    print(f'{item[0]}\t{item[1]}')
    most_freq_radicals[item] = num+1

freq_chars_by_rad_count_stats = {}
practical_chars = set()
char_components = {}
radicals = rads.radical_list

char_set_by_radical = {}
for current in radicals:
    char_set_by_radical.setdefault(current, set())

for i in range(1, 6):
    words = chars_by_rad_count[i]
    freq_chars_by_rad_count_stats[i] = words
    practical_chars.update(words)

    for word in words:
        char_components.setdefault(word, [])

        components = char_dict[word]['components']

        for comp in components:
            if not rads.is_radical_variant(comp):
                continue

            norminal = rads.norminal(comp)
            char_set_by_radical[norminal].add(word)

            char_components[word].append(norminal)

        char_components[word].sort()
        pass
print(freq_chars_by_rad_count_stats.keys())

print(f'{count=}')
print(f'{len(practical_chars)=}')
print(f'{len(char_components)=}')
print(f'{len(char_set_by_radical)=}')


Most freq radical count
3	4268
2	2728
4	2608
5	1004
1	724
6	357
0	149
7	73
8	59
9	22
10	3
Most freq radicals
⼀	2015
⼝	1668
⾋	1222
⼁	1151
⼈	1025
⽇	918
⼆	795
⼂	786
⼋	722
⽉	689
⼗	674
⽊	669
⾦	666
⽔	545
⽷	532
⼃	519
⾔	475
⼑	463
⼄	450
⼇	448
⼤	425
⼩	420
⾍	408
⼼	408
⼌	406
⼟	396
⼿	357
⾩	342
⼜	341
⽥	337
⼧	329
⽵	319
⿃	315
⼓	314
⿂	312
⼍	311
⼉	293
⼥	288
⾙	287
⽬	282
⼛	279
⾡	277
⽯	276
⼡	264
⼚	257
⾐	249
⽕	248
⽲	248
⼹	245
⾞	233
⾜	224
⾺	219
⾫	214
⽰	209
⽟	209
⼽	208
⼘	207
⽁	201
⼫	199
⽶	195
⾨	183
⼭	179
⽹	177
⼐	174
⼅	168
⼨	166
⼲	163
⼒	162
⾴	160
⾌	157
⼏	154
⼠	153
⽻	153
⼱	153
⼙	152
⾷	151
⾬	141
⼔	141
⼦	141
⼎	137
⽴	137
⽿	137
⼕	133
⽝	132
⽺	132
⼯	126
⽖	124
⽩	124
⾣	121
⽌	119
⼞	117
⾅	116
⽫	113
⼴	113
⽳	108
⽧	100
⾰	100
⽋	99
⽤	98
⼺	98
⾑	96
⾀	93
⽜	93
⼸	91
⽄	89
⼰	87
⽢	86
⾈	85
⽅	82
⽎	80
⼣	78
⽮	78
⽼	77
⾖	72
⽭	68
⼪	68
⼻	67
⾤	65
⾽	64
⾒	64
⼬	63
⾥	61
⾗	60
⾟	60
⽍	60
⽐	60
⽽	58
⾮	56
⽂	55
⼖	55
⾭	51
⼶	50
⾛	49
⾓	49
⼳	49
⿅	49
⽸	48
⽈	47
⾕	47
⽒	47
⽚	46
⽱	45
⼮	44
⾉	44
⼵	43
⾘	42
⾻	42
⾳	40
⿒	40
⾱	38
⾏	38
⽏	37
⿓	37
⾃	35
⿁	35
⿈	33
⿇	33
⾚	33

# Build the tree

In [21]:
from treelib import Node, Tree
import itertools


tree = Tree()

base = tree.create_node('', '')

for current in radicals:
    if current in char_set_by_radical:
        char_set = char_set_by_radical[current]

        if char_set:
            node = tree.create_node(identifier=current, tag=current, parent=base, data = char_set)

# tree.show()
# for pre, _, node in RenderTree(root):
#     print("%s%s %i" % (pre, node.name, len(node.chars)))


def add_item(components, current):
    if len(components) < 1:
        return
    
    comp = components[0]

    # print(parent)

    child_comps = current.data - char_set_by_radical[comp]

    if child_comps:
        
        child_id = current.identifier+comp
        
        node=tree.get_node(child_id)

        if not (node):
            node = tree.create_node(identifier=child_id, tag=child_id, parent=current, data=child_comps)
        
        add_item(components[1:], node)

    # tree.show()

count = 0

for word in char_components:
    components = char_components[word]
    if len(char_components[word]) < 2:
        continue
    
    current = tree.get_node(components[0])
    add_item(''.join(components[1:]), current)
    count += 1

    # if count > 500:
    #     break
    pass

tree.show()
tree_string = tree.show(stdout=False)
print(f'{len(tree.nodes)=}')


├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒
│   │   │       └── ⼀⼁⼂⼃⾦
│   │   ├── ⼀⼁⼃
│   │   │   ├── ⼀⼁⼃⼄
│   │   │   │   ├── ⼀⼁⼃⼄⼿
│   │   │   │   ├── ⼀⼁⼃⼄⽅
│   │   │   │   ├── ⼀⼁⼃⼄⽶
│   │   │   │   ├── ⼀⼁⼃⼄⽷
│   │   │   │   ├── ⼀⼁⼃⼄⾡
│   │   │   │   └── ⼀⼁⼃⼄⾩
│   │   │   ├── ⼀⼁⼃⼆
│   │   │   │   ├── ⼀⼁⼃⼆⼈
│   │   │   │   ├── ⼀⼁⼃⼆⼝
│   │   │   │   ├── ⼀⼁⼃⼆⼼
│   │   │   │   ├── ⼀⼁⼃⼆⽇
│   │   │   │   ├── ⼀⼁⼃⼆⽉
│   │   │   │   ├── ⼀⼁⼃⼆⽊
│   │   │   │   ├── ⼀⼁⼃⼆⽕
│   │   │   │   ├── ⼀⼁⼃⼆⽧
│   │   │   │   ├── ⼀⼁⼃⼆⽯
│   │   │   │   ├── ⼀⼁⼃⼆⽰
│   │   │   │   ├── ⼀⼁⼃⼆⽳
│   │   │   │   ├── ⼀⼁⼃⼆⽵
│   │   │   │   ├── ⼀⼁⼃⼆⾈
│   │   │   │   ├── ⼀⼁⼃⼆⾋
│   │   │   │   ├── ⼀⼁⼃⼆⾍
│   │   │   │   ├── ⼀⼁⼃⼆⾔
│   │   │   │   ├── ⼀⼁⼃⼆⾡
│   │   │   │   ├── ⼀⼁⼃⼆⾣
│   │   │   │   ├── ⼀⼁⼃⼆⾩
│   │   │   │   ├── ⼀⼁⼃⼆⿂
│   │   │   │   └── ⼀⼁⼃⼆⿒
│   │   │   └── ⼀⼁⼃⼗
│   │   ├── ⼀⼁⼆
│   │   │   ├── ⼀⼁⼆⼈
│   │   │   │   ├── ⼀⼁⼆

In [22]:
components

['⼁', '⼅', '⼇', '⼋', '⾦']

# Alternative chars for box drawing

'├' => '＼'
'└' => '＼'
'─' => '⼀'
' ' => '　' 

```
Current
＼⼀⼀　⼀
｜　　　＼⼀⼀　⼀⼁
｜　　　｜　　　＼⼀⼀　⼀⼁⼂
｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼂⼃
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⼝
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⽫
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⾒
｜　　　｜　　　｜　　　　　　　＼⼀⼀　⼀⼁⼂⼃⾦
｜　　　｜　　　＼⼀⼀　⼀⼁⼃
｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄
｜　　　｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄⼿
｜　　　｜　　　｜　　　｜　　　＼⼀⼀　⼀⼁⼃⼄⽅

# Originals
├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒
│   │   │       └── ⼀⼁⼂⼃⾦
│   │   ├── ⼀⼁⼃
│   │   │   ├── ⼀⼁⼃⼄
│   │   │   │   ├── ⼀⼁⼃⼄⼿
│   │   │   │   ├── ⼀⼁⼃⼄⽅
```

# Reformat box drawing to fullwidth

In [23]:
def get_chars_from_rads(radicals):
    if not radicals:
        return None

    keys = list(radicals)
    lenkey = len(keys)


    char_set = char_set_by_radical[keys[0]]
    
    # print(f'{len(char_set)=}')

    for key in keys[1:]:
        char_set = char_set & char_set_by_radical[key]

        # print(f'{len(char_set)=}')

    return char_set

# Replacement for fullwidth box drawing characters 
def hex_show(char):
    return '\\u' + hex(ord(char))[2:].upper()

def replace_box_chars(tree):
    new_tree = tree
    char_map = {
        '├' : '＼',
        '└' : '＼',
        '─' : '⼀',
        ' ' : '　',
    }

    for char in char_map:
        new_tree = new_tree.replace(char, char_map[char])

    return new_tree

tree_str = '''
├── ⼀
│   ├── ⼀⼁
│   │   ├── ⼀⼁⼂
│   │   │   └── ⼀⼁⼂⼃
│   │   │       ├── ⼀⼁⼂⼃⼝
│   │   │       ├── ⼀⼁⼂⼃⽫
│   │   │       ├── ⼀⼁⼂⼃⾒'''


# print(f'[├] => [＼] {hex_show('├')} => {hex_show('＼')}')
# print(f'[└] => [＼] {hex_show('└')} => {hex_show('＼')}')
# print(f'[─] => [⼀] {hex_show('─')} => {hex_show('⼀')}')
# print(f'[ ] => [　] {hex_show(' ')} => {hex_show('　')}')

# print(f'tree: \n{tree_str}\nnew tree: \n{replace_box_chars(tree_str)}')



# Build the dictionary

In [24]:
lines = tree_string.split('\n')
key_to_sorted_chars = {}

def replace_key(match_obj):
    key = match_obj.group(0)

    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]

    return f'{key[-1]}: {''.join(chars[:10])} {len(chars)}' 

def replace_simple_key(match_obj):
    key = match_obj.group(0)

    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]

    return f'{key[-1]}: {''.join(chars[:10])} {len(chars)}' 

# Build map between keys and chars at a node
for key in tree.nodes:
   
    if key not in key_to_sorted_chars:
        chars = sort_by_freq(get_chars_from_rads(key))
        key_to_sorted_chars[key] = chars
    else:
        chars = key_to_sorted_chars[key]
    pass
pass

HOME_PAGE = "偏旁"
HOME_PAGE_PY = "piānpáng"
HOME_PAGE1 = HOME_PAGE+'1'
HOME_PAGE2 = HOME_PAGE+'2'

string = ''
HOME_PAGE = "偏旁"
HOME_PAGE_PY = "piānpáng"

back_home = f"""{pleco_make_dark_gray('HOMEPAGE')} {PC_RIGHT_TRIANGLE} {pleco_make_link(HOME_PAGE)}\n"""

fpleco = open('radical_lookup_pleco.txt', 'w', encoding='utf-8')
string+=back_home

def get_rad_pinyin(rad_list):
    return ' '.join([rads.lookup(rad)['pinyin'] for rad in rad_list])

def get_rad_pinyin_abbrev(rad_list):
    return ' '.join([rads.lookup(rad)['pinyin'][0] for rad in rad_list])

MAX_SHOW = 5
WIDE_SPACE = '\u3000'

def node_to_pleco(current):
    children = sorted([x.identifier for x in tree.children(current)])
    current_chars = key_to_sorted_chars[current]
    len_current = len(current)
    
    string = ''
    
    if not current: # This is root
        key_perms = [HOME_PAGE]
        key_py_perms = [HOME_PAGE_PY]
    else:
        key_perms = [''.join(item) for item in itertools.permutations(current)]
        key_py_perms = [get_rad_pinyin(key) for key in key_perms]

    # string = f'\t{key_py}\t'

    if current:
        string += back_home

    if current_chars:
        only_these_radicals = [char for char in current_chars if len(char_components[char]) == len_current]
        count_str = ''
        to_show = len(only_these_radicals) if not children else MAX_SHOW # If no more child branches, show everything

        if to_show > MAX_SHOW:
            count_str = f'+{to_show-MAX_SHOW}'

        shown_chars = [pleco_make_link(char) for char in only_these_radicals[:to_show]]
        num_chars = pleco_make_dark_gray(f'ONLY THESE RADICALS: {len(only_these_radicals)}')

        if len(current_chars) == 1:
            string += f'{num_chars}{f'{PC_MIDDLE_DOT}'.join(shown_chars)}\n'
        else:
            string += f'{num_chars}\n{f'{PC_MIDDLE_DOT}'.join(shown_chars)} {count_str}\n'
        tree
    if children:
        string += pleco_make_dark_gray(f'BRANCHES: {len(children)}\n')

        for child in children:
            if not child:
                continue

            child_chars = key_to_sorted_chars[child]
            child_children = sorted([x.identifier for x in tree.children(child)])

            child_to_show = len(child_chars) if not child_children else MAX_SHOW # If no more child branches, show everything

            shown_chars = [pleco_make_link(char) for char in child_chars[:child_to_show]]
            
            if len(child_chars) == 1:
                actual_components = ''.join(char_components[child_chars[0]])
                string += f'{pleco_make_link(actual_components)}{WIDE_SPACE}{f'{PC_MIDDLE_DOT}'.join(shown_chars)}\n'
            else:
                string += f'{pleco_make_link(child)}{WIDE_SPACE}{f'{PC_MIDDLE_DOT}'.join(shown_chars)} ({len(child_children)}/{len(child_chars)})\n'

    string = string.replace('\n', PC_NEW_LINE)
    
    for key, py in zip(key_perms, key_py_perms):
        fpleco.write(f'{key}\t{py}\t{string}\n')

    for child in children:
        if not child:
            continue

        node_to_pleco(child)

node_to_pleco('')

string = string.replace('\n', PC_NEW_LINE)
fpleco.write(f'{string}\n')

fpleco.close()

In [25]:
# nodes = tree.nodes

count = 0
total = set()
list_py = []
set_py = set()
abb_list_py = []
abb_set_py = set()

for key in tree.nodes:
    if not key:
        continue

    if len(key) < 2:
        continue
    #  if len(tree.get_node(key).data) != 1:
    #     continue

    py = get_rad_pinyin(key)
    list_py.append(py)
    set_py.add(py)

    abb_py = get_rad_pinyin_abbrev(key)
    abb_list_py.append(abb_py)
    abb_set_py.add(abb_py)

    # if py == 'yī jīn':
    #   print(key)
    #   print(tree.get_node(key).data)

print(f"{len(list_py)}")
print(f"{len(abb_set_py)}")

print(f"{len(abb_list_py)}")
print(f"{len(abb_set_py)}")
# print(f'{len(total)}')


def list_duplicates(seq):
    seen = set()
    seen_add = seen.add
    # adds all elements it doesn't know yet to seen and all other to seen_twice
    seen_twice = set(x for x in seq if x in seen or seen_add(x))
    # turn the set into a list (as requested)
    return list(seen_twice)


dupl = list_duplicates(list_py)
print(dupl)  # yields [1, 2, 5]
print(f"{len(dupl)}")

abb_dupl = list_duplicates(abb_list_py)
print(abb_dupl)  # yields [1, 2, 5]
print(f"{len(abb_dupl)}")


8256
4475
8256
4475
['hǎn shān', 'shì mì', 'shì wǎng', 'xiǎo yù', 'mù yán', 'shí jīn', 'yī rì máo', 'jīn fù', 'gān cǎo', 'yī rén kǒu shì', 'rì mù lǎo', 'piě xiǎo mù', 'yī gǔn èr cùn jīn', 'kǒu lì', 'zhǔ jī jīn', 'gōng yuè', 'shān yuè', 'gē shǒu shí', 'mián gǔ', 'mián chén', 'yī shí mù', 'mù yáng', 'shí shān rì lì', 'bā sī mù', 'bā dāo hǎn mù yán', 'mù zhuī', 'gǔn gōng mù', 'zhǔ rén yǔ', 'yī bǔ yòu xiǎo mù', 'yī shí', 'tóu bā shí', 'jīn yuè', 'bā dāo mù', 'kǒu shì', 'yī kǒu shì', 'bǐ shí', 'hǎn mù', 'shí shì mù', 'qiàn shí', 'xī shí', 'shí kǒu shì', 'gǔ niǎo', 'shí shì', 'mù jīn', 'jīn lì', 'shí shì mù cǎo', 'tóu bā mì', 'shān jīn', 'tǔ zhǐ', 'fāng cǎo', 'mù zhú', 'shí mǐ', 'xiǎo shān', 'pū mù', 'kǒu yǎn yù', 'hǎn yòu mù', 'dāo jīn', 'jī shān', 'yī bā jīn', 'rén mù', 'rén shān', 'nǚ wú', 'gǔn mù', 'gōng shí', 'rén shì', 'lì zhú', 'mì kǒu tǔ xiǎo mù', 'mì mù', 'piě shǒu', 'shí shān rì', 'ér bīng mù', 'mù gèn', 'mù lì', 'mì shì', 'lì niǎo', 'rén fāng', 'èr rén mì', 'yī wéi', 'pū mù ěr', '

In [26]:
print('\n'.join(abb_list_py))

q s
b b
j y
s q
g y
r j
j c
z y
s c
r c
x y
è t
t j
l n
g g
y s
z m
l q
d s
x m
h y
m j
j y
s w
j y
g f
s m
g c
g y
r l
y c
n w
m y
z s
y d
f j
r y
s l
l m
m m
s k
d m
s r
y x
q y
g c
y ě
x t
k m
c f
j x
p s
b f
j y
b y
y è
p y
r l
y l
g z
t z
x y
s c
x x
y b
m h
y r
y f
n m
t y
c y
t m
t s
g s
m s
p j
x j
j q
y c
r l
x n
t h
y y
g z
p y
l t
b m
é f
d n
k t
y f
g j
n d
y y
s y
y c
h s
p m
p r
b k
h y
j l
h q
y m
m y
h w
y m
è f
g s
p r
c y
g b
p g
s h
g j
ě s
c s
d n
t s
r z
y y
s y
j j
z m
b c
y g
n y
m c
y y
b s
p y
z y
s l
g ě
y j
p f
n g
f s
y c
c z
f m
j s
g m
j m
q y
y c
z n
m b
m y
s h
y n
s g
z j
g n
g n
x g
y j
m s
c f
y s
m y
m y
r j
b s
y m
c s
d j
d b
s x
s q
p z
s y
f m
t s
y x
r c
r j
c y
g x
y l
c g
k f
g y
x s
g z
r s
c m
p c
t j
t c
b z
g s
f x
z z
g c
s f
y m
c q
s m
x x
z h
w m
s c
g k
w c
g ě
k n
y h
m g
d j
t y
k m
g y
m m
z r
s m
r n
f j
p y
g f
p c
t c
z l
c f
m p
g c
j c
m m
s d
h h
y z
y q
g y
z d
c f
p d
p z
d y
h z
q b
x t
y s
b z
j z
r z
y l
y h
p s
b f
j c


In [27]:
print('\n'.join(list_py))

qiàn shí
bǔ bèi
jǐ yǒu
shí qīng
gān yú
rén jiù
jǐ cǎo
zhú yuè
shí chuò
rén chuò
xiǎo yuè
èr tǔ
tǔ jǐ
lì niǎo
gān gǔ
yī shí
zhú mǎ
lì qīng
dà shì
xuán mì
hū yán
mù jīn
jīn yè
shǒu wú
jué yè
gōng fù
shǒu mù
gān chóng
gān yán
rì lǎo
yòng chóng
nǚ wú
mì yán
zhǔ shì
yī dòu
fāng jīn
rén yòng
sī lì
lì mǐ
mì má
shí kǒu
dà mǎ
shí rì
yuè xíng
quǎn yù
gōng chì
yòu ěr
xīn tián
kǒu mù
chē fēi
jīn xué
piě shí
bèi fù
jī yòu
bāo yán
yī èr
piě yè
rén lì
yǔ lì
gǔn zhú
tián zú
xué yīn
sè chì
xī xué
yuè bèi
mián huǒ
yī rì
yán fēng
nǚ mǐ
tián yǔ
chì yè
tián mì
tóu shì
gōng shuǐ
mù shuǐ
piě jīn
xī jīn
jīn qiàn
yuè chuò
rén lì
xué niǎo
tǔ huǒ
yī yú
gèn zú
piàn yú
lì tián
bǔ mù
ér fù
dà niǎo
kǒu tǔ
yī fù
gǔn jì
nè dòu
yuè yī
shān yuè
yǐ chē
hū shǐ
pū mì
piě rì
bāo kǒu
hé yú
jīn lì
hé qí
yī mù
mián yuè
huǒ wéi
yù mǎ
èr fù
gǔn shū
piě rén
cùn yǒu
gān bí
piě gé
shān hé
gèn jīn
ěr shí
chuò shǒu
dà nǚ
tǔ shǒu
rén zǐ
yè yuè
shū yáng
jiōng jīn
zǐ mǐ
bāo cǎo
yī guǐ
nǚ yáng
mù chóng
yǒu yǔ
bīng shí
pí yán
zhǔ yán
shì 

# Distributions of radicals
| Len | Count | Pecentitle |
|-----|-------|------------|
| 0   | 149   | 1.2%       |
| 1   | 724   | 7.3%       |
| 2   | 2728  | 30.0%      |
| 3   | 4268  | 65.6%      |
| 4   | 2608  | 87.3%      |
| 5   | 1004  | 95.7%      |
| 6   | 357   | 98.7%      |
| 7   | 73    | 99.3%      |
| 8   | 59    | 99.8%      |
| 9   | 22    | 100.0%     |
| 10  | 3     | 100.0%     |
| Sum | 11995 |


So we will focus on words with 5 or less radicals, remove words without radicals

# Most frequent rads

| Pos | Radical | Char count | Percentitle |
|-----|---------|------------|-------------|
| 1   | ⼀       | 2015       | 5.3%        |
| 2   | ⼝       | 1668       | 9.7%        |
| 3   | ⾋       | 1222       | 13.0%       |
| 4   | ⼁       | 1151       | 16.0%       |
| 5   | ⼈       | 1025       | 18.7%       |
| 6   | ⽇       | 918        | 21.2%       |
| 7   | ⼆       | 795        | 23.3%       |
| 8   | ⼂       | 786        | 25.4%       |
| 9   | ⼋       | 722        | 27.3%       |
| 10  | ⽉       | 689        | 29.1%       |
| 11  | ⼗       | 674        | 30.9%       |
| 12  | ⽊       | 669        | 32.6%       |
| 13  | ⾦       | 666        | 34.4%       |
| 14  | ⽔       | 545        | 35.8%       |
| 15  | ⽷       | 532        | 37.3%       |
| 16  | ⼃       | 519        | 38.6%       |
| 17  | ⾔       | 475        | 39.9%       |
| 18  | ⼑       | 463        | 41.1%       |
| 19  | ⼄       | 450        | 42.3%       |
| 20  | ⼇       | 448        | 43.5%       |
| 21  | ⼤       | 425        | 44.6%       |
| 22  | ⼩       | 420        | 45.7%       |
| 23  | ⾍       | 408        | 46.8%       |
| 24  | ⼼       | 408        | 47.9%       |
| 25  | ⼌       | 406        | 49.0%       |
| 26  | ⼟       | 396        | 50.0%       |
| 27  | ⼿       | 357        | 50.9%       |
| 28  | ⾩       | 342        | 51.9%       |
| 29  | ⼜       | 341        | 52.8%       |
| 30  | ⽥       | 337        | 53.6%       |
| 31  | ⼧       | 329        | 54.5%       |
| 32  | ⽵       | 319        | 55.4%       |
| 33  | ⿃       | 315        | 56.2%       |
| 34  | ⼓       | 314        | 57.0%       |
| 35  | ⿂       | 312        | 57.9%       |
| 36  | ⼍       | 311        | 58.7%       |
| 37  | ⼉       | 293        | 59.4%       |
| 38  | ⼥       | 288        | 60.2%       |
| 39  | ⾙       | 287        | 61.0%       |
| 40  | ⽬       | 282        | 61.7%       |
| 41  | ⼛       | 279        | 62.5%       |
| 42  | ⾡       | 277        | 63.2%       |
| 43  | ⽯       | 276        | 63.9%       |
| 44  | ⼡       | 264        | 64.6%       |
| 45  | ⼚       | 257        | 65.3%       |
| 46  | ⾐       | 249        | 66.0%       |
| 47  | ⽕       | 248        | 66.6%       |
| 48  | ⽲       | 248        | 67.3%       |
| 49  | ⼹       | 245        | 67.9%       |
| 50  | ⾞       | 233        | 68.5%       |
| 51  | ⾜       | 224        | 69.1%       |
| 52  | ⾺       | 219        | 69.7%       |
| 53  | ⾫       | 214        | 70.3%       |
| 54  | ⽟       | 209        | 70.8%       |
| 55  | ⽰       | 209        | 71.4%       |
| 56  | ⼽       | 208        | 71.9%       |
| 57  | ⼘       | 207        | 72.5%       |
| 58  | ⽁       | 201        | 73.0%       |
| 59  | ⼫       | 199        | 73.5%       |
| 60  | ⽶       | 195        | 74.1%       |
| 61  | ⾨       | 183        | 74.5%       |
| 62  | ⼭       | 179        | 75.0%       |
| 63  | ⽹       | 177        | 75.5%       |
| 64  | ⼐       | 174        | 75.9%       |
| 65  | ⼅       | 168        | 76.4%       |
| 66  | ⼨       | 166        | 76.8%       |
| 67  | ⼲       | 163        | 77.3%       |
| 68  | ⼒       | 162        | 77.7%       |
| 69  | ⾴       | 160        | 78.1%       |
| 70  | ⾌       | 157        | 78.5%       |
| 71  | ⼏       | 154        | 78.9%       |
| 72  | ⽻       | 153        | 79.3%       |
| 73  | ⼱       | 153        | 79.7%       |
| 74  | ⼠       | 153        | 80.1%       |
| 75  | ⼙       | 152        | 80.5%       |
| 76  | ⾷       | 151        | 80.9%       |
| 77  | ⼔       | 141        | 81.3%       |
| 78  | ⾬       | 141        | 81.7%       |
| 79  | ⼦       | 141        | 82.1%       |
| 80  | ⼎       | 137        | 82.4%       |
| 81  | ⽿       | 137        | 82.8%       |
| 82  | ⽴       | 137        | 83.2%       |
| 83  | ⼕       | 133        | 83.5%       |
| 84  | ⽝       | 132        | 83.9%       |
| 85  | ⽺       | 132        | 84.2%       |
| 86  | ⼯       | 126        | 84.5%       |
| 87  | ⽩       | 124        | 84.9%       |
| 88  | ⽖       | 124        | 85.2%       |
| 89  | ⾣       | 121        | 85.5%       |
| 90  | ⽌       | 119        | 85.8%       |
| 91  | ⼞       | 117        | 86.1%       |
| 92  | ⾅       | 116        | 86.4%       |
| 93  | ⽫       | 113        | 86.7%       |
| 94  | ⼴       | 113        | 87.0%       |
| 95  | ⽳       | 108        | 87.3%       |
| 96  | ⽧       | 100        | 87.6%       |
| 97  | ⾰       | 100        | 87.9%       |
| 98  | ⽋       | 99         | 88.1%       |
| 99  | ⽤       | 98         | 88.4%       |
| 100 | ⼺       | 98         | 88.6%       |
| 101 | ⾑       | 96         | 88.9%       |
| 102 | ⽜       | 93         | 89.1%       |
| 103 | ⾀       | 93         | 89.4%       |
| 104 | ⼸       | 91         | 89.6%       |
| 105 | ⽄       | 89         | 89.9%       |
| 106 | ⼰       | 87         | 90.1%       |
| 107 | ⽢       | 86         | 90.3%       |
| 108 | ⾈       | 85         | 90.5%       |
| 109 | ⽅       | 82         | 90.8%       |
| 110 | ⽎       | 80         | 91.0%       |
| 111 | ⼣       | 78         | 91.2%       |
| 112 | ⽮       | 78         | 91.4%       |
| 113 | ⽼       | 77         | 91.6%       |
| 114 | ⾖       | 72         | 91.8%       |
| 115 | ⼪       | 68         | 92.0%       |
| 116 | ⽭       | 68         | 92.1%       |
| 117 | ⼻       | 67         | 92.3%       |
| 118 | ⾤       | 65         | 92.5%       |
| 119 | ⾒       | 64         | 92.7%       |
| 120 | ⾽       | 64         | 92.8%       |
| 121 | ⼬       | 63         | 93.0%       |
| 122 | ⾥       | 61         | 93.2%       |
| 123 | ⽐       | 60         | 93.3%       |
| 124 | ⾟       | 60         | 93.5%       |
| 125 | ⽍       | 60         | 93.6%       |
| 126 | ⾗       | 60         | 93.8%       |
| 127 | ⽽       | 58         | 93.9%       |
| 128 | ⾮       | 56         | 94.1%       |
| 129 | ⽂       | 55         | 94.2%       |
| 130 | ⼖       | 55         | 94.4%       |
| 131 | ⾭       | 51         | 94.5%       |
| 132 | ⼶       | 50         | 94.6%       |
| 133 | ⿅       | 49         | 94.8%       |
| 134 | ⾛       | 49         | 94.9%       |
| 135 | ⼳       | 49         | 95.0%       |
| 136 | ⾓       | 49         | 95.2%       |
| 137 | ⽸       | 48         | 95.3%       |
| 138 | ⾕       | 47         | 95.4%       |
| 139 | ⽒       | 47         | 95.5%       |
| 140 | ⽈       | 47         | 95.7%       |
| 141 | ⽚       | 46         | 95.8%       |
| 142 | ⽱       | 45         | 95.9%       |
| 143 | ⼮       | 44         | 96.0%       |
| 144 | ⾉       | 44         | 96.1%       |
| 145 | ⼵       | 43         | 96.3%       |
| 146 | ⾘       | 42         | 96.4%       |
| 147 | ⾻       | 42         | 96.5%       |
| 148 | ⿒       | 40         | 96.6%       |
| 149 | ⾳       | 40         | 96.7%       |
| 150 | ⾏       | 38         | 96.8%       |
| 151 | ⾱       | 38         | 96.9%       |
| 152 | ⽏       | 37         | 97.0%       |
| 153 | ⿓       | 37         | 97.1%       |
| 154 | ⿁       | 35         | 97.2%       |
| 155 | ⾃       | 35         | 97.3%       |
| 156 | ⿈       | 33         | 97.4%       |
| 157 | ⿇       | 33         | 97.4%       |
| 158 | ⾚       | 33         | 97.5%       |
| 159 | ⽾       | 32         | 97.6%       |
| 160 | ⿊       | 32         | 97.7%       |
| 161 | ⽦       | 31         | 97.8%       |
| 162 | ⼷       | 31         | 97.9%       |
| 163 | ⾄       | 30         | 97.9%       |
| 164 | ⽪       | 30         | 98.0%       |
| 165 | ⾆       | 30         | 98.1%       |
| 166 | ⽑       | 29         | 98.2%       |
| 167 | ⾼       | 28         | 98.3%       |
| 168 | ⾠       | 28         | 98.3%       |
| 169 | ⽣       | 27         | 98.4%       |
| 170 | ⾵       | 27         | 98.5%       |
| 171 | ⾝       | 24         | 98.5%       |
| 172 | ⿏       | 24         | 98.6%       |
| 173 | ⽙       | 24         | 98.7%       |
| 174 | ⽓       | 24         | 98.7%       |
| 175 | ⽀       | 23         | 98.8%       |
| 176 | ⿀       | 23         | 98.8%       |
| 177 | ⿑       | 23         | 98.9%       |
| 178 | ⽨       | 22         | 99.0%       |
| 179 | ⽞       | 22         | 99.0%       |
| 180 | ⾇       | 21         | 99.1%       |
| 181 | ⾂       | 21         | 99.1%       |
| 182 | ⾧       | 21         | 99.2%       |
| 183 | ⾲       | 20         | 99.2%       |
| 184 | ⿎       | 18         | 99.3%       |
| 185 | ⽘       | 18         | 99.3%       |
| 186 | ⽠       | 17         | 99.4%       |
| 187 | ⿐       | 16         | 99.4%       |
| 188 | ⽆       | 16         | 99.5%       |
| 189 | ⽃       | 15         | 99.5%       |
| 190 | ⽡       | 15         | 99.6%       |
| 191 | ⿌       | 15         | 99.6%       |
| 192 | ⿕       | 14         | 99.6%       |
| 193 | ⽛       | 14         | 99.7%       |
| 194 | ⿆       | 14         | 99.7%       |
| 195 | ⾁       | 14         | 99.7%       |
| 196 | ⾎       | 12         | 99.8%       |
| 197 | ⽗       | 11         | 99.8%       |
| 198 | ⾊       | 10         | 99.8%       |
| 199 | ⾸       | 10         | 99.9%       |
| 200 | ⾹       | 9          | 99.9%       |
| 201 | ⾪       | 8          | 99.9%       |
| 202 | ⾯       | 7          | 99.9%       |
| 203 | ⾢       | 6          | 99.9%       |
| 204 | ⿍       | 5          | 99.9%       |
| 205 | ⼊       | 5          | 100.0%      |
| 206 | ⿄       | 4          | 100.0%      |
| 207 | ⿋       | 4          | 100.0%      |
| 208 | ⿔       | 3          | 100.0%      |
| 209 | ⾿       | 2          | 100.0%      |
| 210 | ⾶       | 1          | 100.0%      |
| 211 | ⿉       | 1          | 100.0%      |
| 212 | ⼾       | 1          | 100.0%      |
|     |         | 37789      |

